src/renderer/features/balances/model/balance-subscription-model.ts

```diff
@@ -21,52 +24,73 @@ type SubscribeParams = {
   apis: Record<ChainId, ApiPromise>;
   statuses: Record<ChainId, ConnectionStatus>;
   subscriptions: BalanceSubscribeMap;
-  accountIds: AccountId[];
+  accounts: Account[];
 };
 const createSubscriptionsBalancesFx = createEffect(
-  ({ chains, apis, statuses, subscriptions, accountIds }: SubscribeParams): Record<ChainId, SubscriptionObject> => {
+  async ({
+    chains,
+    apis,
+    statuses,
+    subscriptions,
+    accounts,
+  }: SubscribeParams): Promise<Record<ChainId, SubscriptionObject>> => {
     const newSubscriptions = {} as Record<ChainId, SubscriptionObject>;
 
-    Object.entries(apis).forEach(async ([chainId, api]) => {
-      const networkConnected = statuses[chainId as ChainId] === ConnectionStatus.CONNECTED;
-      const oldSubscription = subscriptions[chainId as ChainId];
-
-      if (!networkConnected) {
-        await unsubscribeBalancesFx({
-          chainId: chainId as ChainId,
-          subscription: oldSubscription,
-        });
-
-        return;
-      }
-
-      if (oldSubscription) {
-        const sameAccounts = oldSubscription?.accounts.join(',') === accountIds.join(',');
-
-        if (sameAccounts) {
-          return;
+    await Promise.all(
```

Comment:
Reduce nesting please

Reply:
Overall this effect looks really fragile
It creates subscription which is fine but all those unsubs 
Please add `TODO` with a [task](https://app.clickup.com/t/86939u5ut) to revisit this effect in future